src/fondue/Fondue.js

```diff
@@ -649,60 +649,93 @@ export default class Fondue {
 	get featureChars() {
 		const { cmap, GSUB } = this._font.opentype.tables;
 
+		// Human readable names for lookup types
+		const lookupTypes = {
+			1: "Single Substitution",
+			2: "Multiple Substitution",
+			3: "Alternate Substitution",
+			4: "Ligature Substitution",
+			5: "Contextual Substitution",
+			6: "Chained Contexts Substitution",
+			7: "Extension Substitution",
+			8: "Reverse Chaining Contextual Single Substitution",
+		};
+
 		if (!GSUB) return {};
 
 		function letterFor(glyphid) {
 			return cmap.reverse(glyphid).unicode;
 		}
 
-		function parseLookup(lookup, script, lang, feature, currentAllGlyphs) {
-			if (!(feature.featureTag in currentAllGlyphs)) {
-				currentAllGlyphs[feature.featureTag] = {
-					type: lookup.lookupType,
-					input: [],
-					alternateCount: [],
-				};
+		// [1,2,3] + [3,4,5] = [1,2,3,4,5]
+		function mergeUniqueCoverage(existing, addition) {
+			return [...new Set([...(existing || []), ...addition])];
+		}
+
+		// Returns glyphs that are mapped directly to characters for
+		// this coverage. If a glyph maps to another glyph, it's
+		// ignored.
+		// If only a specific rangeRecord needs to be processed, e.g.
+		// for lookup type 3, you can pass the desired index.
+		function charactersFromGlyphs(coverage, index) {
```

Comment:
Could we extract all these inner functions into private functions on Fondue.js? Or perhaps extract the entire type 6 lookup to a separate file? This function is pretty big now (~ 300 lines) and it feels like there are a lot of details that are not really important to understanding the flow of the type 6 lookup. E.g. it's good to know reading this code that it's looking up chars from glyphs, but _how_ that's done is not really relevant when you try to just understand the type 6 lookup, right?